src/test/web/smoke.test/index.ts

```diff
@@ -1,29 +1,64 @@
-// imports mocha for the browser, defining the `mocha` global.
-require('mocha/mocha');
+// Re-export extension entry point, so that the output from this file
+// when bundled can be used as entry point for extension as well as tests.
+// The same objects/types will be used as the module is only ever loaded once by nodejs.
+import * as extension from '../../../extension.web';
+import * as vscode from 'vscode';
+import type { IExtensionApi } from '../../../platform/api';
+import type { IExtensionContext } from '../../../platform/common/types';
+import { IExtensionTestApi } from '../../common';
+import { JVSC_EXTENSION_ID } from '../../../platform/common/constants';
 
-export function run(): Promise<void> {
-    return new Promise((c, e) => {
-        mocha.setup({
-            ui: 'tdd',
-            reporter: undefined
-        });
+let activatedResponse: undefined | IExtensionApi;
 
-        // bundles all files in the current directory matching `*.test`
-        const importAll = (r: __WebpackModuleApi.RequireContext) => r.keys().forEach(r);
-        importAll(require.context('.', true, /\.web.test$/));
+// Basically this is the entry point for the extension.
```

Comment:
This bundle will be the entry point for the extension, and everything uses the same bundle and tests are in here too.